compiler-cli/src/fs.rs

```diff
@@ -342,10 +342,34 @@ fn is_gleam_path(path: &Utf8Path, dir: impl AsRef<Utf8Path>) -> bool {
     )
 }
 
+fn is_gleam_build_dir(e: &ignore::DirEntry) -> Option<bool> {
+    if !e.path().is_dir() || !e.path().ends_with("build") {
+        return Some(false);
+    }
+
+    let has_gleam_toml_as_sibiling = e
+        .path()
+        .parent()?
+        .read_dir()
+        .ok()?
+        .any(|e| e.unwrap().file_name().to_str() == Some("gleam.toml"));
```

Comment:
This will iterate over the whole directory again. We can avoid that work by doing something like this pseudocode:

```rust
e.path().parent().join("gleam.toml").is_file()
```

Reply:
Thanks! I missed that :)